src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -532,10 +546,10 @@ public void RegisterConnectResult(SocketError error)
             }
         }
 
-        private bool TryBeginOperation<TOperation>(ref OperationQueue<TOperation> queue, TOperation operation, Interop.Sys.SocketEvents events, bool maintainOrder, out bool isStopped)
+        private bool TryBeginOperation<TOperation>(ref OperationQueue<TOperation> queue, object queueLock, TOperation operation, Interop.Sys.SocketEvents events, bool maintainOrder, out bool isStopped)
```

Comment:
Passing around a lock to use like this makes me a bit nervous.  Would it make sense to instead just say that TryBeginOperation (maybe renamed TryBeginOperationUnlocked) doesn't do any locking and it's up to the caller to ensure the right lock is held?  In a bunch of the call sites, it appears that'll already be the case.

Otherwise, we might want to add a bunch of asserts to verify that locks aren't being taken in a bad order.